pkg/project/util/util.go

```diff
@@ -37,11 +38,32 @@ func Finalized(namespace *kapi.Namespace) bool {
 }
 
 // Finalize will remove the origin finalizer from the namespace
-func Finalize(kubeClient kclient.Interface, namespace *kapi.Namespace) (*kapi.Namespace, error) {
+func Finalize(kubeClient kclient.Interface, namespace *kapi.Namespace) (result *kapi.Namespace, err error) {
 	if Finalized(namespace) {
 		return namespace, nil
 	}
-	return finalizeInternal(kubeClient, namespace, false)
+
+	// there is a potential for a resource conflict with base kubernetes finalizer
+	// as a result, we handle resource conflicts in case multiple finalizers try
+	// to finalize at same time
+	retry := true
+	for retry {
+		result, err = finalizeInternal(kubeClient, namespace, false)
+		if err == nil {
+			retry = false
+		} else {
```

Comment:
All the same comments I made upstream.  Which means you get double the
review and have double the review debt.

On Sep 21, 2015, at 4:55 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/project/util/util.go
<https://github.com/openshift/origin/pull/4752#discussion_r40024677>:

>  	if Finalized(namespace) {
>  		return namespace, nil
>  	}
> -	return finalizeInternal(kubeClient, namespace, false)
> +
> +	// there is a potential for a resource conflict with base kubernetes finalizer
> +	// as a result, we handle resource conflicts in case multiple finalizers try
> +	// to finalize at same time
> +	retry := true
> +	for retry {
> +		result, err = finalizeInternal(kubeClient, namespace, false)
> +		if err == nil {
> +			retry = false
> +		} else {

can unnest the else if we break in the above case

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4752/files#r40024677>.